File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.go
func numberOfSubarrays(nums []int, k int) int {
    return countSubarrays(nums, k) - countSubarrays(nums, k - 1)
}

func countSubarrays(nums []int, k int) int {
    left, right, oddCount, count := 0, 0, 0, 0
    for right < len(nums) {
        oddCount += (nums[right] & 1)
        for oddCount > k {
            oddCount -= (nums[left] & 1)
            left++
        }
        count += right - left + 1
        right++
    }
    return count
} 

File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var numberOfSubarrays = function(nums, k) {
    return countSubarrays(nums, k) - countSubarrays(nums, k - 1)
};

function countSubarrays(nums, k) {
    let left = 0, right = 0, oddCount = 0, count = 0;
    while (right < nums.length) {
        oddCount += (nums[right] & 1);
        while (oddCount > k) {
            oddCount -= (nums[left] & 1);
            left++;
        }
        count += right - left + 1;
        right++;
    }
    return count;
} 